### ISSUE TITLE
raise_delivery_errors doesn’t work

### ISSUE THREAD

**Issue by PeterMozesMerl** [rails/rails#35330]
### Steps to reproduce
1. Set config.action_mailer.raise_delivery_errors = true in production.rb

(Rails 5.2.2)

### Expected behavior
The application raises delivery errors

### Actual behavior
The application doesn’t raise delivery errors

### Description
The setting isn’t called raise_delivery_errors_for_stmp.

The documentation says: "Whether or not errors should be raised if the email fails to be delivered. This only works if the external email server is configured for immediate delivery".

I consider the :sendmail setting and the deliver_now as an instant delivery attempt. The common sense also says that deliver_now should not swallow easy to catch errors. For example:

```ruby
	config.action_mailer.raise_delivery_errors = true
	config.action_mailer.delivery_method = :sendmail
	config.action_mailer.sendmail_settings = { location: '/i/excel/in/hiding' }
```

$ rails console -e production

```
2.6.0 :001 > u = User.find 1
 => #<User id: 1...>

2.6.0 :002 > UserMailer.confirm_email(u).deliver_now
  Rendering user_mailer/confirm_email.html.slim within layouts/mailer
  Rendered user_mailer/confirm_email.html.slim within layouts/mailer (12.6ms)
  Rendering user_mailer/confirm_email.text.slim within layouts/mailer
  Rendered user_mailer/confirm_email.text.slim within layouts/mailer (1.9ms)
UserMailer#confirm_email: processed outbound mail in 35.8ms
sh: /i/excel/in/hiding: No such file or directory
Sent mail to xxx@xxx.com (11.9ms)
Date: Tue, 19 Feb 2019 23:28:00 +0100
[...]
Mime-Version: 1.0
Content-Type: multipart/alternative;
[...]
 => #<Mail::Message:70316996859100, Multipart: true, Headers: ...>
```

I almost deployed it to production. Sure, I can triple check if the binary exists on every server. But if ActionMailer doesn’t raise an exception when the binary is missing, I have no reason to believe it would ever raise anything.

It’s unfortunate because the :sendmail method with Exim is between one and two magnitudes faster (on my servers) than either Exim or FreeBSD’s sendmail sitting on localhost:25 even without SSL/TLS.

Bonus: it’s counter-intuitive that deliver_now! is meant to swallow exceptions while ActiveModel, since eternity, has been raising exceptions with bang methods.

Anyway, the main problem is that the :sendmail settings doesn’t seem to care at all.

**Comment by y-yagi** [comment]
It seems already fixed with https://github.com/mikel/mail/commit/7e1196bd29815a0901d7290c82a332c0959b163a. Please try with the master branch of `mail` gem. 
If not fixed the problem, please create an issue to https://github.com/mikel/mail/issues. Thanks!

**Comment by PeterMozesMerl** [comment]
Thanks. Also, as soon as I have the fix in stable, I will do detailed testing of the rest of the possible errors when the method is :sendmail.

**Comment by PeterMozesMerl** [comment]
Although I am not much into putting an edge version into production. I tried. The master version can’t even send mail regardless of whether it is :smtp or :sendmail

```
gem 'mail', git: 'https://github.com/mikel/mail.git', ref: 'ff17438f2af361bf0c20e897945e2a88a0a6b283'
```

```
2.6.0 :005 > m.deliver_now
Sent mail to xxx@xxx.com (1.9ms)
nil
Traceback (most recent call last):
        2: from (irb):5
        1: from (irb):5:in `rescue in irb_binding'
NoMethodError (undefined method `ascii_only?' for nil:NilClass)
```
I understand it’s an upstream issue. I will create the ticket there. However, the fix in the mail gem is 3.5 years old and still unreleased. Its master branch is not working, at least not with Ruby 2.6 and Rails 5.2.2.

In its current state, the :sendmail delivery method is broken.
